reverse_proxy.go

```diff
@@ -371,6 +371,25 @@ func (p *ReverseProxy) CheckHardTimeoutEnforced(spec *APISpec, req *http.Request
 	return false, globalConf.ProxyDefaultTimeout
 }
 
+func (p *ReverseProxy) CheckHeaderAllowed(hdr string, spec *APISpec, req *http.Request) bool {
```

Comment:
If feel like "allowed" is not the right word here, because we talking about headers which should be removed.

Reply:
Yeah, I'm struggling a little bit with this one.
I wanted to keep consistent with other similar methods which are all called `Check<...>`.
But something like `CheckHeaderInRemoveList` means it will have to be negated in most cases (ie `if !CheckHeaderInRemoveList...`) which some people are a bit uncomfortable with (I personally have no problems with that).

I'm committing it as `CheckHeaderInRemoveList`, but name suggestions here are very welcome :smile: